### PR TITLE
Pass Client IP address to MinIO on x-forwarded-for header

### PR DESCRIPTION
**Pull request by dvaldivia**
This PR introduces a custom `RoundTripper` for the transport we use when communicating with MinIO via the SDKs.

The intention is to communicate the IP address of the actual user to MinIO.

While reviewing this PR:

If the request was available, that was used as an inline solution, like when the client is instantiated from the handler
`context.Context` was used whenever was possible as the middleware can introduce the IP address and have it be pass it around
websocket can't use the same `context.Context` have their own remoteAddress call

Fixes https://github.com/minio/console/issues/2290

### CODE REVIEW

File: restapi/ws_handle.go [hunk 1]
```diff
@@ -91,6 +92,7 @@ type WSConn interface {
 	writeMessage(messageType int, data []byte) error
 	close() error
 	readMessage() (messageType int, p []byte, err error)
+	remoteAddress() string
```
Comment (cesnietor) [inline restapi/ws_handle.go:95]: Can we add this one to `mockConn` so that it passes the lint errors.

Comment (dvaldivia) [reply restapi/ws_handle.go:95]: added

File: cmd/console/app_commands.go [hunk 2]
```diff
@@ -41,8 +41,8 @@ func StartServer(ctx *cli.Context) error {
 	}
 
 	xctx := context.Background()
-	transport := restapi.PrepareSTSClientTransport(false)
-	if err := logger.InitializeLogger(xctx, transport); err != nil {
+	transport := restapi.PrepareSTSClientTransport(false, "127.0.0.1")
```
Comment (reivaj05) [inline cmd/console/app_commands.go:44]: Could you add "127.0.0.1" as constant in `const.go` file?

**Review by reivaj05 — CHANGES_REQUESTED** [review]

File: cmd/console/app_commands.go (hunk 2 above)
Comment (dvaldivia) [reply cmd/console/app_commands.go:44]: done

**Review by bexsoft — APPROVED** [review]

File: restapi/tls.go [hunk 3]
```diff
@@ -23,8 +23,19 @@ import (
 	"time"
 )
 
+type ConsoleTransport struct {
+	Transport *http.Transport
+	ClientIP  string
+}
+
+func (t *ConsoleTransport) RoundTrip(req *http.Request) (*http.Response, error) {
+	req.Header.Add("X-Forwarded-For", t.ClientIP)
```
Comment (vadmeste) [inline restapi/tls.go:32]: Do we really need to use X-Forwarded-For header? it is a known header that has its own purpose and Console is not a real proxy, I guess we should be able to use a different header name without issues.

Comment (dvaldivia) [reply restapi/tls.go:32]: we are technically proxying requests for the users via the UI, and we need to communicate to MinIO the real IP address of the client behind the browser and MinIO is already reading this header by default so I don't think we should be using a custom header

**Review by reivaj05 — APPROVED** [review]

**Pull request merged**